Solutions/Corvus.Azure.Storage.Tenancy/Microsoft/Extensions/DependencyInjection/TenancyBlobStorageServiceCollectionExtensions.cs

```diff
@@ -30,48 +29,33 @@ public static class TenancyBlobStorageServiceCollectionExtensions
         }
 
         /// <summary>
-        /// Adds services required by tenancy Azure storage based stores, and configures the default
-        /// tenant's default storage account settings based on configuration settings.
+        /// Add components for constructing tenant-specific blob storage containers.
         /// </summary>
-        /// <param name="services">The service collection.</param>
+        /// <param name="services">The target service collection.</param>
         /// <param name="getOptions">Function to get the configuration options.</param>
-        /// <returns>The modified service collection.</returns>
+        /// <returns>The service collection.</returns>
         public static IServiceCollection AddTenantCloudBlobContainerFactory(
-            this IServiceCollection services,
-            Func<IServiceProvider, TenantCloudBlobContainerFactoryOptions> getOptions)
+            this IServiceCollection services, Func<IServiceProvider,
+            TenantCloudBlobContainerFactoryOptions> getOptions)
         {
+            if (services is null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
             if (services.Any(s => typeof(ITenantCloudBlobContainerFactory).IsAssignableFrom(s.ServiceType)))
             {
                 return services;
             }
 
             services.AddRootTenant();
-
-            services.AddTenantCloudBlobContainerFactory(
-                (sp, rootTenant) =>
+            services.AddTransient<BlobStorageConfiguration>();
```

Comment:
This (and its equivalents on the other tenanted storage providers) confuses me. What's depending on this thing being available via DI? Aren't all `BlobStorageConfiguration` instances now for specific `BlobStorageContainerDefinition`s on specific tenants?

Reply:
Yes, I had intended to remove these but only got rid of one.

Reply:
Yeah, I should have removed that.